{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# The host cell's ER proteostasis network profoundly shapes the protein sequence space accessible to HIV envelope"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Overview\n",
    "\n",
    "The first part of this analysis processes FASTQ files generated by Barcoded-subamplicon sequencing and converts them to codon counts, which are the frequencies of mutations at each site of Env. The analysis primarily uses [dms2_batch_bcsubamp](https://jbloomlab.github.io/dms_tools2/dms2_batch_bcsubamp.html).\n",
    "\n",
    "The second part of this analysis calculates [differential selection](https://jbloomlab.github.io/dms_tools2/diffsel.html#diffsel) (diffsel) from codoncounts.csv files. Briefly, differential selection quantifies the relative enrichment of each amino acid substitution at a given site in the selection condition compared to the mock condition. The analysis primarily uses [dms2_batch_diffsel](https://jbloomlab.github.io/dms_tools2/dms2_batch_diffsel.html)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import modules and define general variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using dms_tools2 version 2.4.10\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import sys\n",
    "import glob\n",
    "import pandas\n",
    "from IPython.display import IFrame\n",
    "from IPython.display import display, HTML\n",
    "import dms_tools2\n",
    "import dms_tools2.plot\n",
    "import dms_tools2.sra\n",
    "import dms_tools2.utils\n",
    "import dms_tools2.diffsel\n",
    "from dms_tools2.ipython_utils import showPDF\n",
    "\n",
    "print(\"Using dms_tools2 version {0}\".format(dms_tools2.__version__))\n",
    "\n",
    "# results will go in this directory\n",
    "resultsdir = './results/' \n",
    "if not os.path.isdir(resultsdir):\n",
    "    os.mkdir(resultsdir)\n",
    "    \n",
    "# number of CPUs to use; use all available\n",
    "ncpus = -1\n",
    "\n",
    "# use existing results if available\n",
    "use_existing = 'yes'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define the samples\n",
    "\n",
    "We define samples needed for processing the Barcoded-subamplicon deep sequencing data. We first create a list of sampleIDs, which are the names of the FASTQ files. We then create a list of sampleNames, which are short descriptions of the FASTQ files and also what the output file names will be based on. Last, we create a list of R1 (forward read) and R2 (reverse read) files.\n",
    "\n",
    "We use the following abbreviations to name our samples:\n",
    "\n",
    " * Mt1, Mt2, and Mt3 are used to indicate biological replicate 1, 2, and 3, respectively\n",
    "\n",
    " * D, T, DT, and V are used to indicate dox-treated (+XBP1s), TMP-treated (+ATF6), dox- and TMP-treated (+XBP1s/+ATF6). and vehicle-treated (Basal) samples, respectively\n",
    "\n",
    " * pLAI is used to indicate p0 mutant viral library\n",
    "\n",
    " * WT is used to indicate wild-type Env"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>sampleID</th>\n",
       "      <th>name</th>\n",
       "      <th>R1</th>\n",
       "      <th>R2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193001</td>\n",
       "      <td>Mt1-V</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193002</td>\n",
       "      <td>Mt1-D</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193003</td>\n",
       "      <td>Mt1-T</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193004</td>\n",
       "      <td>Mt1-DT</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193005</td>\n",
       "      <td>Mt2-V</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193006</td>\n",
       "      <td>Mt2-D</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193007</td>\n",
       "      <td>Mt2-T</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193008</td>\n",
       "      <td>Mt2-DT</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193009</td>\n",
       "      <td>Mt3-V</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193010</td>\n",
       "      <td>Mt3-D</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193011</td>\n",
       "      <td>Mt3-T</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193012</td>\n",
       "      <td>Mt3-DT</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193013</td>\n",
       "      <td>Wt-V</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193014</td>\n",
       "      <td>pLAI-Wt</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193015</td>\n",
       "      <td>pLAI-Mt1</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193016</td>\n",
       "      <td>pLAI-Mt2</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>171013Sho_D17-193017</td>\n",
       "      <td>pLAI-Mt3</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Directory for FASTQ files\n",
    "fastqdir = '/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_DMS_EN'\n",
    "\n",
    "# List of sampleIDs\n",
    "sampleIDs = ['171013Sho_D17-193001','171013Sho_D17-193002','171013Sho_D17-193003',\n",
    "             '171013Sho_D17-193004','171013Sho_D17-193005','171013Sho_D17-193006',\n",
    "             '171013Sho_D17-193007','171013Sho_D17-193008','171013Sho_D17-193009',\n",
    "             '171013Sho_D17-193010','171013Sho_D17-193011','171013Sho_D17-193012',\n",
    "             '171013Sho_D17-193013','171013Sho_D17-193014','171013Sho_D17-193015',\n",
    "             '171013Sho_D17-193016','171013Sho_D17-193017']\n",
    "\n",
    "# List of sampleNames\n",
    "sampleNames = ['Mt1-V','Mt1-D','Mt1-T','Mt1-DT','Mt2-V','Mt2-D','Mt2-T','Mt2-DT',\n",
    "                   'Mt3-V','Mt3-D','Mt3-T','Mt3-DT','Wt-V','pLAI-Wt','pLAI-Mt1','pLAI-Mt2','pLAI-Mt3']\n",
    "\n",
    "# List of R1 and R2 files\n",
    "r1filename = []\n",
    "r2filename = []\n",
    "\n",
    "\n",
    "for i in range(len(sampleIDs)):\n",
    "    sampleID = sampleIDs[i]\n",
    "    r1filename.append('/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_DMS_EN/'+sampleID+'_1_sequence.remerged.fastq')\n",
    "    r2filename.append('/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_DMS_EN/'+sampleID+'_2_sequence.remerged.fastq')\n",
    "\n",
    "samples = pandas.DataFrame(list(zip(sampleIDs, sampleNames, r1filename, r2filename)),\n",
    "               columns =['sampleID', 'name', 'R1', 'R2'])\n",
    "\n",
    "display(HTML(samples.to_html(index=False)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run dms2_batch_bcsubamp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Here is the batch file that we write to CSV format to use as input:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>name</th>\n",
       "      <th>R1</th>\n",
       "      <th>R2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Mt1-V</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt1-D</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt1-T</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt1-DT</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt2-V</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt2-D</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt2-T</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt2-DT</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt3-V</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt3-D</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt3-T</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Mt3-DT</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Wt-V</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>pLAI-Wt</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>pLAI-Mt1</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>pLAI-Mt2</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>pLAI-Mt3</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "      <td>/Users/sjhendel/Documents/FASTQ_Files/HIV_Env_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# file containing wildtype LAI Env sequence\n",
    "refseq = './HIV_LAI_Env_reference_seq.fa'\n",
    "# Specify R1 and R2 read lengths and gene range of each subamplicon for dms_barcodedsubamplicons\n",
    "alignspecs_ori = ['103,336,45,42',\n",
    "                  '337,561,45,42',\n",
    "                  '520,795,43,36',\n",
    "                  '772,1002,42,42',\n",
    "                  '994,1221,42,39',\n",
    "                  '1186,1422,45,39',\n",
    "                  '1414,1653,39,39',\n",
    "                  '1651,1902,36,42',\n",
    "                  '1897,2136,36,36']\n",
    "\n",
    "alignspecs = ' '.join(alignspecs_ori)\n",
    "alignspecs = alignspecs.strip()\n",
    "\n",
    "# raw counts and alignments placed in this directory\n",
    "rawcountsdir = os.path.join(resultsdir, 'rawcodoncounts')\n",
    "if not os.path.isdir(rawcountsdir):\n",
    "    os.mkdir(rawcountsdir)\n",
    "    \n",
    "# write sample information to a batch file for dms2_batch_bcsubamplicons\n",
    "countsbatchfile = os.path.join(rawcountsdir, 'batch.csv')\n",
    "print(\"Here is the batch file that we write to CSV format to use as input:\")\n",
    "display(HTML(samples[['name', 'R1', 'R2']].to_html(index=False)))\n",
    "samples[['name', 'R1', 'R2']].to_csv(countsbatchfile, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now running dms2_batch_bcsubamp...\n",
      "Completed dms2_batch_bcsubamp.\n"
     ]
    }
   ],
   "source": [
    "print('\\nNow running dms2_batch_bcsubamp...')\n",
    "log = !dms2_batch_bcsubamp \\\n",
    "            --batchfile {countsbatchfile} \\\n",
    "            --refseq {refseq} \\\n",
    "            --alignspecs {alignspecs} \\\n",
    "            --outdir {rawcountsdir} \\\n",
    "            --summaryprefix summary \\\n",
    "            --R1trim {200} \\\n",
    "            --R2trim {160} \\\n",
    "            --fastqdir {fastqdir} \\\n",
    "            --ncpus {ncpus} \\\n",
    "            --use_existing {use_existing} \n",
    "print(\"Completed dms2_batch_bcsubamp.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove sites that are not mutagenized\n",
    "\n",
    "In the Env mutant libraries we used, the N-terminal signal peptide and the C-terminal cytoplasmic tail were excluded from mutagenesis owing to their dramatic impact on Env expression and/or HIV infectivity. The .codoncounts files were manually modified to exclude these sites (sites 0-34, 708-722) and saved in the directory './results/codoncounts'."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Here are the modified codoncounts files\n",
      "./results/codoncounts/Mt1-D_codoncounts.csv\n",
      "./results/codoncounts/MT1-DT_codoncounts.csv\n",
      "./results/codoncounts/Mt1-T_codoncounts.csv\n",
      "./results/codoncounts/Mt1-V_codoncounts.csv\n",
      "./results/codoncounts/Mt2-D_codoncounts.csv\n",
      "./results/codoncounts/Mt2-DT_codoncounts.csv\n",
      "./results/codoncounts/Mt2-T_codoncounts.csv\n",
      "./results/codoncounts/Mt2-V_codoncounts.csv\n",
      "./results/codoncounts/Mt3-D_codoncounts.csv\n",
      "./results/codoncounts/Mt3-DT_codoncounts.csv\n",
      "./results/codoncounts/Mt3-T_codoncounts.csv\n",
      "./results/codoncounts/Mt3-V_codoncounts.csv\n",
      "./results/codoncounts/pLAI-Mt1_codoncounts.csv\n",
      "./results/codoncounts/pLAI-Mt2_codoncounts.csv\n",
      "./results/codoncounts/pLAI-Mt3_codoncounts.csv\n",
      "./results/codoncounts/pLAI-Wt_codoncounts.csv\n",
      "./results/codoncounts/Wt-V_codoncounts.csv\n"
     ]
    }
   ],
   "source": [
    "countsdir = './results/codoncounts'\n",
    "countsfiles = os.path.join(countsdir, '*')\n",
    "\n",
    "print(\"Here are the modified codoncounts files\")\n",
    "\n",
    "for name in glob.glob(countsfiles):\n",
    "    print(name)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run dms2_batch_diffsel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Here is the batch file that we write to CSV format to use as input:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>group</th>\n",
       "      <th>name</th>\n",
       "      <th>sel</th>\n",
       "      <th>mock</th>\n",
       "      <th>err</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>XBP1s</td>\n",
       "      <td>replicate1</td>\n",
       "      <td>Mt1-D_codoncounts.csv</td>\n",
       "      <td>Mt1-V_codoncounts.csv</td>\n",
       "      <td>Wt-V_codoncounts.csv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATF6</td>\n",
       "      <td>replicate1</td>\n",
       "      <td>Mt1-T_codoncounts.csv</td>\n",
       "      <td>Mt1-V_codoncounts.csv</td>\n",
       "      <td>Wt-V_codoncounts.csv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XBPsATF6</td>\n",
       "      <td>replicate1</td>\n",
       "      <td>Mt1-DT_codoncounts.csv</td>\n",
       "      <td>Mt1-V_codoncounts.csv</td>\n",
       "      <td>Wt-V_codoncounts.csv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XBP1s</td>\n",
       "      <td>replicate2</td>\n",
       "      <td>Mt2-D_codoncounts.csv</td>\n",
       "      <td>Mt2-V_codoncounts.csv</td>\n",
       "      <td>Wt-V_codoncounts.csv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATF6</td>\n",
       "      <td>replicate2</td>\n",
       "      <td>Mt2-T_codoncounts.csv</td>\n",
       "      <td>Mt2-V_codoncounts.csv</td>\n",
       "      <td>Wt-V_codoncounts.csv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XBPsATF6</td>\n",
       "      <td>replicate2</td>\n",
       "      <td>Mt2-DT_codoncounts.csv</td>\n",
       "      <td>Mt2-V_codoncounts.csv</td>\n",
       "      <td>Wt-V_codoncounts.csv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XBP1s</td>\n",
       "      <td>replicate3</td>\n",
       "      <td>Mt3-D_codoncounts.csv</td>\n",
       "      <td>Mt3-V_codoncounts.csv</td>\n",
       "      <td>Wt-V_codoncounts.csv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ATF6</td>\n",
       "      <td>replicate3</td>\n",
       "      <td>Mt3-T_codoncounts.csv</td>\n",
       "      <td>Mt3-V_codoncounts.csv</td>\n",
       "      <td>Wt-V_codoncounts.csv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XBPsATF6</td>\n",
       "      <td>replicate3</td>\n",
       "      <td>Mt3-DT_codoncounts.csv</td>\n",
       "      <td>Mt3-V_codoncounts.csv</td>\n",
       "      <td>Wt-V_codoncounts.csv</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# diffsel files placed in this directory\n",
    "diffseldir = os.path.join(resultsdir, 'diffsel')\n",
    "if not os.path.isdir(diffseldir):\n",
    "    os.mkdir(diffseldir)\n",
    "\n",
    "diffselbatchfile = os.path.join(diffseldir, 'batch.csv')\n",
    "\n",
    "# Specify which samples to use as mock condition and selection condition to calculate differential selection\n",
    "diffselbatch = pandas.DataFrame.from_records([\n",
    "    ('XBP1s', 'replicate1', 'Mt1-D_codoncounts.csv', 'Mt1-V_codoncounts.csv'),\n",
    "    ('ATF6', 'replicate1', 'Mt1-T_codoncounts.csv', 'Mt1-V_codoncounts.csv'),\n",
    "    ('XBPsATF6', 'replicate1', 'Mt1-DT_codoncounts.csv', 'Mt1-V_codoncounts.csv'),\n",
    "    ('XBP1s', 'replicate2', 'Mt2-D_codoncounts.csv', 'Mt2-V_codoncounts.csv'),\n",
    "    ('ATF6', 'replicate2', 'Mt2-T_codoncounts.csv', 'Mt2-V_codoncounts.csv'),\n",
    "    ('XBPsATF6', 'replicate2', 'Mt2-DT_codoncounts.csv', 'Mt2-V_codoncounts.csv'),\n",
    "    ('XBP1s', 'replicate3', 'Mt3-D_codoncounts.csv', 'Mt3-V_codoncounts.csv'),\n",
    "    ('ATF6', 'replicate3', 'Mt3-T_codoncounts.csv', 'Mt3-V_codoncounts.csv'),\n",
    "    ('XBPsATF6', 'replicate3', 'Mt3-DT_codoncounts.csv', 'Mt3-V_codoncounts.csv'),\n",
    "    ],\n",
    "    columns = ['group', 'name', 'sel', 'mock']\n",
    "    )\n",
    "\n",
    "# Specify which sample to use as error correction\n",
    "diffselbatch['err'] = 'Wt-V_codoncounts.csv'\n",
    "\n",
    "print(\"Here is the batch file that we write to CSV format to use as input:\")\n",
    "display(HTML(diffselbatch.to_html(index=False)))\n",
    "diffselbatch.to_csv(diffselbatchfile, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now running dms2_batch_diffsel...\n",
      "Completed dms2_batch_bcsubamp.\n"
     ]
    }
   ],
   "source": [
    "print('\\nNow running dms2_batch_diffsel...')\n",
    "log = !dms2_batch_diffsel \\\n",
    "        --summaryprefix summary \\\n",
    "        --batchfile {diffselbatchfile} \\\n",
    "        --outdir {diffseldir} \\\n",
    "        --indir {countsdir} \\\n",
    "        --use_existing {use_existing}\n",
    "print(\"Completed dms2_batch_bcsubamp.\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
